### ISSUE TITLE
Decode.getBytes does not clear output buffer

### ISSUE THREAD

**Issue by normcf** [brendan-duncan/archive#30]
I'm trying to use Decode by adding bytes (Deflate.addBytes) as they are generated with FLUSH_PARTIAL and pull bytes off with Deflate.getBytes.  When I call Deflate.addBytes then Deflate.getBytes the second time I get the bytes from the first call PLUS the new bytes added to the output buffer.  I think, you should either have:

`_output.clear()` in Deflate.getBytes (after grabbing the bytes to return), or add some new function e.g.

```
List<int> takeBytes() {
    _flushPending();
    List<int> bytes = _output.getBytes();
    _output.clear();
    return bytes;
}
```

I used this function and my code works.

FYI: I am not doing the whole file in memory because the data are too large.  As I take the output buffer, I write into a temporary File in the browser and create a url for the user to download.

**Comment by brendan-duncan** [comment]
Thanks, I'll look into adding this as soon as I get a chance.

**Comment by normcf** [comment]
While you're in there I noticed in gzip_encoder.dart the constant OS_UNKNOWN is 25, but according to http://www.zlib.org/rfc-gzip.html it's 255.  It probably doesn't matter, but should be an easy fix once the code is opened up.  Take your time, I have a workaround for now by editing in the packages directory.  It'll get written over on the next pug get.

**Comment by brendan-duncan** [comment]
I haven't forgotten about this, I just haven't had much personal programming time lately.

**Comment by brendan-duncan** [comment]
Finally got around to this, sorry it took so long. Decided against modifying getBytes and added the extra takeBytes method. Published in 1.0.28.
